sqlalchemy-stubs/engine/reflection.pyi

```diff
@@ -1,69 +1,83 @@
 from typing import Any
+from typing import Collection
+from typing import Dict
+from typing import List
 from typing import Optional
+from typing import Set
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
 
-from .base import Connectable as Connectable
-from .base import Connection as Connection
-from .base import Engine as Engine
-from .. import exc as exc
-from .. import inspection as inspection
-from .. import sql as sql
-from .. import util as util
-from ..sql import operators as operators
-from ..sql.type_api import TypeEngine as TypeEngine
-from ..util import topological as topological
+from .base import Connectable
+from ..sql.schema import Table
 
-def cache(fn: Any, self: Any, con: Any, *args: Any, **kw: Any): ...
+_TInspector = TypeVar("_TInspector", bound=Inspector)
+
+def cache(fn: Any, self: Any, con: Any, *args: Any, **kw: Any) -> Any: ...
 
 class Inspector:
-    def __init__(self, bind: Any): ...
+    def __init__(self, bind: Connectable): ...
     @classmethod
-    def from_engine(cls, bind: Any): ...
+    def from_engine(
+        cls: Type[_TInspector], bind: Connectable
+    ) -> _TInspector: ...
     @property
-    def default_schema_name(self): ...
-    def get_schema_names(self): ...
-    def get_table_names(self, schema: Optional[Any] = ...): ...
-    def has_table(self, table_name: Any, schema: Optional[Any] = ...): ...
+    def default_schema_name(self) -> str: ...
+    def get_schema_names(self) -> List[str]: ...
+    def get_table_names(self, schema: Optional[str] = ...) -> List[str]: ...
+    def has_table(
+        self, table_name: str, schema: Optional[str] = ...
+    ) -> bool: ...
     def has_sequence(
-        self, sequence_name: Any, schema: Optional[Any] = ...
-    ): ...
-    def get_sorted_table_and_fkc_names(self, schema: Optional[Any] = ...): ...
-    def get_temp_table_names(self): ...
-    def get_temp_view_names(self): ...
+        self, sequence_name: str, schema: Optional[str] = ...
+    ) -> bool: ...
+    def get_sorted_table_and_fkc_names(
+        self, schema: Optional[str] = ...
+    ) -> List[Tuple[Optional[str], List[Tuple[str, str]]]]: ...
```

Comment:
python generic code syncacs is really ugly compared to other languages like ts :(

Reply:
It can be, but even in TS you can do `Array<Array<string | null, Array<Array<string, string>>>>`

Reply:
sure, hopefully 3.9+ is nice since you can use lowercase stuff and `|` for unions

Reply:
Using `|` for unions and `Optional` is such a huge improvement